# HG changeset patch
# User David Bateman <dbateman@free.fr>
# Date 1288712768 -3600
#      Tue Nov 02 16:46:08 2010 +0100
# Node ID 30bcd1aa9f09fe4d27dbae7b1da212536b12d9aa
# Parent  2271261f088ae8df22b08d6d3ad3d9d7cede3423
Trivial patch to previous change

diff --git a/src/ls-hdf5.cc b/src/ls-hdf5.cc
--- a/src/ls-hdf5.cc
+++ b/src/ls-hdf5.cc
@@ -666,21 +666,21 @@ hdf5_add_scalar_attr (hid_t loc_id, hid_
 {
   herr_t retval = 0;
 
   hid_t as_id = H5Screate (H5S_SCALAR);
 
   if (as_id >= 0)
     {
 #if HAVE_HDF5_18
-      hid_t a_id = H5Acreate (loc_id, attr_name, H5T_NATIVE_UCHAR, 
+      hid_t a_id = H5Acreate (loc_id, attr_name, type_id, 
                               as_id, H5P_DEFAULT, H5P_DEFAULT);
 #else
       hid_t a_id = H5Acreate (loc_id, attr_name,
-                              H5T_NATIVE_UCHAR, as_id, H5P_DEFAULT);
+                              type_id, as_id, H5P_DEFAULT);
 #endif
       if (a_id >= 0)
         {
           retval = H5Awrite (a_id, type_id, buf);
 
           H5Aclose (a_id);
         }
       else
@@ -689,22 +689,16 @@ hdf5_add_scalar_attr (hid_t loc_id, hid_
       H5Sclose (as_id);
     }
   else
     retval = as_id;
 
   return retval;
 }
 
-
-
-
-
-
-
 // Save an empty matrix, if needed. Returns
 //    > 0  Saved empty matrix
 //    = 0  Not an empty matrix; did nothing
 //    < 0  Error condition
 int
 save_hdf5_empty (hid_t loc_id, const char *name, const dim_vector d)
 {
   hsize_t sz = d.length ();
